### PR TITLE
Charter proposal for infrastructure lifecycle wg

### PR DESCRIPTION
**Pull request by GenPage**
This PR includes the content of the Charter definition for the Infrastrucutre Lifecycle Working Group. The content was discussed in the following google doc: https://docs.google.com/document/d/1eK7m13F8SHuL2ji0U__5S_dlD2UBX0POquj6AtBWkoc/edit

@lianmakesthings @thschue @roberthstrand Let me know what you think

### CODE REVIEW

**Comment by bschaatsbergen** [comment]
Yay! 🙌🏼

**Comment by roberthstrand** [comment]
Charter looks good. I would like to have a sync on how we should go about selecting the chairs for the working group, so that is part of request to the TOC.

DCO also needs to be resolved. So if you could follow this [recipe](https://github.com/cncf/tag-app-delivery/pull/691/checks?check_run_id=26328806136) @GenPage, that would be awesome.

**Comment by GenPage** [comment]
@roberthstrand Sounds good, I'll put it on the agenda for this friday. Also rebased to fixed the DCO check.

**Comment by roberthstrand** [comment]
Thank you @GenPage, I will bring this along to the TOC.

File: infra-lifecycle-wg/charter/charter.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+## Alignment with TAG Runtime Charter
+
+We partner with the Runtime TAG to ensure that cloud-native runtime environments provide the fundamental workload execution abstractions and mechanisms needed to integrate with infrastructure lifecycle management tools. This collaboration focuses on guaranteeing that both runtime and infrastructure layers work cohesively to deliver applications effectively.
+
+## Working mode/expected outcome
+
+The Infrastructure Lifecycle Working Group will achieve its goals through regular meetings ([Zoom](https://zoom-lfx.platform.linuxfoundation.org/meeting/96148400770?password=767d45df-c7cf-4400-9239-e789115cc85e&invite=true)), open communication channels ([#wg-infrastructure-lifecycle](https://cloud-native.slack.com/archives/C06USDTN683) on Slack), and collaboration with other TAGs. This will result in a standardised approach for managing the lifecycle of infrastructure in a cloud-native setting, develop a set of comprehensive best practices, and publish valuable resources for the cloud-native community around infrastructure lifecycle management
```
Comment (linsun) [inline infra-lifecycle-wg/charter/charter.md:84]: Hi, could you specify what is the success criteria of the WG?  I'd like to understand when we should continue or discontinue the WG.

Comment (GenPage) [reply infra-lifecycle-wg/charter/charter.md:84]: Hello @linsun, I'm not sure if this is something that we intend to define in the charter itself as I haven't seen examples of this in other WG charters. Do you have any examples? I happy to bring to up for discussion in our next WG meeting.

Comment (GenPage) [reply infra-lifecycle-wg/charter/charter.md:84]: @linsun Actually, @TheFoxAtWork left some excellent comments about the stated goals of the WG in the charter that I think we could clarify and address, which would also address your question. Please let me know what you think.

https://docs.google.com/document/d/1eK7m13F8SHuL2ji0U__5S_dlD2UBX0POquj6AtBWkoc/edit?disco=AAABOxARaAk

Comment (linsun) [reply infra-lifecycle-wg/charter/charter.md:84]: Thank you, I will follow up in the google doc :)

**Review by abangser — APPROVED** [review]
> Originally, the Platform Working Group focused on clarifying the coordination between application development and infrastructure delivery. While this remains important, the group's focus has evolved to consider platforms as complete products. This broader perspective necessitates a renewed emphasis on the underlying infrastructure abstractions that provide the essential building blocks for these platform offerings

Love this! I am very excited to have more WGs allowing for more deep investigation into the pieces of successful cloud native adoption/delivery. 

Once all accepted, let's find time to talk about some of the more infra focused open topics raised over the last couple of years in the platform WG and see if they fit with your early work.

File: infra-lifecycle-wg/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+## Chairs
+
+- Dylan Page (@genpage)
+- Ryan Nowak (@rynowak)
+- Bruno Schaatsbergen (@bschaatsbergen)
+- Jochen Zehnder (@elft3r)
+
+## Meetings
+
+* Meeting schedule: Bi-weekly on Friday at [1600 UTC](https://dateful.com/convert/utc?t=16)
```
[the remark this answers is not in the document]
Comment (bschaatsbergen) [reply infra-lifecycle-wg/README.md]: Thank you for flagging this @abangser, I've addressed it in 84997c4.

File: infra-lifecycle-wg/charter/charter.md (hunk 1 above)
Comment (bschaatsbergen) [reply infra-lifecycle-wg/charter/charter.md]: We have significantly narrowed the scope of this working group to focus on a single deliverable: _develop a framework for managing the lifecycle of infrastructure in public, private, or hybrid cloud environments based on cloud-native principles. This framework will guide end-users in managing infrastructure to ensure it is secure, resilient, manageable, sustainable, and observable._

**Comment by bschaatsbergen** [comment]
We've addressed comments and feedback, and the charter has been updated.

**Review by TheFoxAtWork — COMMENTED** [review]
LGTM

File: infra-lifecycle-wg/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+To contribute please join our Slack channel and/or biweekly meetings.
+
+#wg-infrastructure-lifecycle Slack channel: <https://cloud-native.slack.com/archives/C06USDTN683>. Get an invite at <https://slack.cncf.io/>
+
+## Chairs
+
+- Bruno Schaatsbergen (@bschaatsbergen)
+- Dylan Page (@genpage)
+- Jochen Zehnder (@elft3r)
+- Ryan Nowak (@rynowak)
```
Comment (linsun) [inline infra-lifecycle-wg/README.md:14]: Is there a reason why we need to start with 4 co-chairs?  We'd recommend 2-3 in general as working through a leadership draft doc.

File: infra-lifecycle-wg/charter/charter.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+While savvy users are embracing cloud native practices, infrastructure requirements are inherently diverse. However, there's an opportunity to champion technology-agnostic best practices. Infrastructure lifecycle management deserves the same level of attention and planning we dedicate to established areas of cloud native development. This ensures security, resilience, manageability, sustainability, and observability. This working group aims to bridge this critical gap by clarifying what a standardised approach for managing the infrastructure lifecycle looks like and develop best practices.
+
+## Deliverable
+
+Develop a framework for managing the lifecycle of infrastructure in public, private, or hybrid cloud environments based on cloud-native principles. This framework will guide end-users in managing infrastructure to ensure it is secure, resilient, manageable, sustainable, and observable.
```
[the remark this answers is not in the document]
Comment (linsun) [reply infra-lifecycle-wg/charter/charter.md:13]: Ok i see, it won't be an OS project below.

File: infra-lifecycle-wg/charter/charter.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+### Infrastructure Tool Orchestration
+
+The growing need for GitOps-style deployments drives demand for orchestrating declarative cloud infrastructure using Infrastructure as Code Tools. These orchestrators manage cloud resources either using native APIs or IaC tools. They offer flexibility across cloud providers and tooling preferences. Tools like TACoS, Crossplane, and AWS Controller for Kubernetes (ACK) exemplify this powerful approach, simplifying cloud infrastructure management.
+
+## Examples of known tools aimed to manage the lifecycle of infrastructure:
+
+Without any claim to comprehensiveness:
+
+* Ansible
+* AWS CloudFormation & AWS CDK
```
Comment (linsun) [inline infra-lifecycle-wg/charter/charter.md:54]: Is this AWS cloud only?

**Review by linsun — COMMENTED** [review]
LGTM, just left 1-2 comments about number of leads and focus of clouds

File: infra-lifecycle-wg/charter/charter.md (hunk 5 above)
Comment (GenPage) [reply infra-lifecycle-wg/charter/charter.md:54]: It's not an exhaustive list. We state on line 51: `Without any claim to comprehensiveness:`

File: infra-lifecycle-wg/README.md (hunk 3 above)
Comment (GenPage) [reply infra-lifecycle-wg/README.md:14]: This was an active discussion with TAG Chairs @roberthstrand and @lianmakesthings. I have already self-nominated for TAG Tech Lead, intending to step down as chair and step in as the TAG Liaison for the WG should I be selected.

In the interim, the current TAG sponsor, @roberthstrand, will be a tie-breaker if needed.

Comment (linsun) [reply infra-lifecycle-wg/README.md:14]: Ok cool, that sounds good, thx for the background info!

**Review by linsun — APPROVED** [review]

**Comment by linsun** [comment]
I approve assuming TAG app delivery co-chairs are also in favor of this cc @lianmakesthings @roberthstrand @thschue

**Review by kgamanji — APPROVED** [review]
LGTM

**Comment by bschaatsbergen** [comment]
> +1 binding 

Thanks, @kgamanji yay! I noticed that your vote isn't reflected on the TOC issue in the binding votes. Just wanted to bring it to your attention.

File: infra-lifecycle-wg/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+The [charter](./charter/charter.md) describes the goals and activites of this group.
+
+To contribute please join our Slack channel and/or biweekly meetings.
+
+#wg-infrastructure-lifecycle Slack channel: <https://cloud-native.slack.com/archives/C06USDTN683>. Get an invite at <https://slack.cncf.io/>
+
+## Chairs
+
+- Bruno Schaatsbergen (@bschaatsbergen)
+- Dylan Page (@genpage)
```
Comment (lianmakesthings) [inline infra-lifecycle-wg/README.md:12]: @GenPage has agreed to withdraw his chair nomination in favour of becoming TAG TL and liaison to the WG

Comment (lianmakesthings) [inline infra-lifecycle-wg/README.md:12]: @GenPage has agreed to withdraw his chair nomination in favour of becoming TAG TL and liaison to the WG

**Review by lianmakesthings — CHANGES_REQUESTED** [review]

File: infra-lifecycle-wg/README.md (hunk 6 above)
Comment (bschaatsbergen) [reply infra-lifecycle-wg/README.md]: Fixed in [ac93684](https://github.com/cncf/tag-app-delivery/pull/691/commits/ac93684540c89c499d27caea964757bb932d9f65)

**Comment by bschaatsbergen** [comment]
Vote passed by the TOC, https://github.com/cncf/toc/issues/1383#issuecomment-2326905473  🎉

**Review by lianmakesthings — APPROVED** [review]

**Pull request merged**